src/diagnostics/diagnostic.jl

```diff
@@ -125,6 +125,28 @@ macro with_error(compute_function_expr)
     )
 end
 
+"""
+    diagnostic_as_vectors(writer::ClimaDiagnostics.DictWriter, diagnostic)
+
+Extract `diagnostic` from given `writer` as tuple of vectors (time and value).
+
+`diagnostic` is typically a string with the short name of the diagnostic.
+"""
+function diagnostic_as_vectors(writer::DictWriter, diagnostic, layer)
```

Comment:
Let's make `layer` a keyword argument that defaults to nothing for diagnostics with no depth.
```suggestion
function diagnostic_as_vectors(writer::DictWriter, diagnostic; layer)
```
Add it to the docstring too

Reply:
good idea!